include/ignition/gazebo/SdfEntityCreator.hh

```diff
@@ -155,6 +155,18 @@ namespace ignition
       /// \param[in] _parent Entity which should be _child's parent.
       public: void SetParent(Entity _child, Entity _parent);
 
+      /// \brief Overloaded function to recursively create model entities
+      /// and make sure a) only one canonical link is created per model tree,
+      /// and b) we override the nested model's static property to true if
+      /// its parent is static
+      /// \param[in] _model SDF model object.
+      /// \param[in] _createCanonicalLink True to create a canonical link
+      /// component and attach to its child link entity
+      /// \param[in] _staticParent True if parent is static, false otherwise.
+      /// \return Model entity.
+      private: Entity CreateEntities(const sdf::Model *_model,
+          bool _createCanonicalLink, bool _staticParent);
```

Comment:
Is static the only option that the nested models will ever override? Maybe consider passing some type that can be extended in the future. A map with key-value pairs? Or maybe it's not worth it...

Reply:
I'm not sure what other params will be added. This function has evolved as I get more info about the nested model proposal. I'm leaning towards leaving as is for now since it's private and we can change it later if we need too.